### PR TITLE
[DOC][MRG] Add glossary

### LINKED ISSUE #75: Create glossary in documentation
Nilearn contains a lot of concepts that need to be explained (e.g. niimg(s)). Adding a glossary to the documentation could be a good idea. 

This is also a mean to improve the naming conventions in the code.

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Fixes #75

### CODE REVIEW

**Comment by NicolasGensollen** [comment]
I think this is ready for reviews. Here is the link to the glossary page ([link](https://6865-1235740-gh.circle-artifacts.com/0/doc/_build/html/glossary.html)), which is also accessible through the nav bar of the website. 
The glossary is far from being finished yet, but adding terms and improving current definitions could be easy first issues.

File: doc/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+.. glossary::
+    :sorted:
+
+    
+    ANOVA
+    Anova
+        Analysis of variance.
+
+    AUC
+        Area under the curve.
```
Comment (bthirion) [inline doc/glossary.rst:21]: Link to sklearn's documentation ?

File: doc/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+If you wish to add a missing term, please `create a new issue`_ or
+`open a Pull Request`_.
+
+.. glossary::
+    :sorted:
+
+    
+    ANOVA
+    Anova
+        Analysis of variance.
```
Comment (bthirion) [inline doc/glossary.rst:18]: wikipedia page ?

File: doc/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+    BIDS
+    bids
+        `Brain Imaging Data Structure`_ is a simple and easy to adopt way
+        of organizing neuroimaging and behavioral data.
+    
+    BOLD
+    Bold
+    bold
+        Blood oxygenation level dependent.
```
Comment (bthirion) [inline doc/glossary.rst:31]:
```suggestion
        Blood oxygenation level dependent. This is the kind of signal measured by functional Magnetic Resonance Imaging.
```

File: doc/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+    CanICA
+        Canonical independant component analysis.
```
Comment (bthirion) [inline doc/glossary.rst:34]: Link to the paper that introduced it ?

File: doc/glossary.rst [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        A `contrast`_ is a linear combination of variables (parameters or
+        statistics) whose coefficients add up to zero, allowing comparison
+        of different treatments.
+
+    Decoding
+        `Decoding`_ consists in predicting, from brain images, the conditions
+        associated to trial.
+
+    EEG
+        Electroencephalography.
```
Comment (bthirion) [inline doc/glossary.rst:47]: Lin to wikipedia paage + add one descritpive sentence ?

File: doc/glossary.rst [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    
+    EPI
+        Echo-Planar Imaging.
```
Comment (bthirion) [inline doc/glossary.rst:50]:
```suggestion
        Echo-Planar Imaging. This is the type of sequence used to acquire functional or diffusion MRI data.
```

File: doc/glossary.rst [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+
+    FDR correction
+        `False discovery rate`_ controlling procedures are designed to control
+        the expected proportion of "discoveries" (rejected null hypotheses)
+        that are false (incorrect rejections of the null).
+
+    FIR
+        Finite impulse response.
```
Comment (bthirion) [inline doc/glossary.rst:58]: This is a type of free-form temporal filter that is used to link neural activity with hemodynamic response, when there is uncertainty on the true model.

File: doc/glossary.rst [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+        oxygen). Oxyhaemoglobin and deoxyhaemoglobin have different magnetic
+        properties (diamagnetic and paramagnetic, respectively), and they
+        affect the local magnetic field in different ways.
+        The signal picked up by the MRI scanner is sensitive to these
+        modifications of the local magnetic field.
+
+    FPR correction
+    FPR
+    fpr
+        False positive rate correction.
```
Comment (bthirion) [inline doc/glossary.rst:76]: Needs a definition

File: doc/glossary.rst [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+        A `functional connectome`_ is a set of connections representing brain 
+        interactions between regions.
+
+    FWER correction
+        `Family-wise error rate`_ is the probability of making one or more
+        false discoveries, or type I errors when performing multiple
+        hypotheses tests.
+
+    GLM
+        General Linear Model.
```
Comment (bthirion) [inline doc/glossary.rst:96]: This is the name of the models traditionally fit to fMRI data, where one linear model is fit to each voxel time course.

File: doc/glossary.rst [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+
+    HRF
+    hrf
+        Haemodynamic response function.
```
Comment (bthirion) [inline doc/glossary.rst:100]:
```suggestion
        Haemodynamic response function. This is a temporal filter that converts neural signal to hemodynamic signals observable with fMRI.
```

File: doc/glossary.rst [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+
+    ICA
+        Independant component analysis.
```
Comment (bthirion) [inline doc/glossary.rst:103]: Link needed, maybe wikipedia or sklearn.

File: doc/glossary.rst [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+
+    MEG
+        Magnetoencephalography.
```
Comment (bthirion) [inline doc/glossary.rst:106]: idem, wikipedia + short sentence

File: doc/glossary.rst [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+
+    MNI
+        MNI stands for "Montreal Neurological Institute". Usually, this is
+        used to reference the MNI space/template. The current standard MNI 
+        template is the ICBM152, which is the average of 152 normal MRI scans 
+        that have been matched to the MNI305 using a 9 parameter affine transform.
+
+    MVPA
+        Mutli-Voxel Pattern Analysis.
```
Comment (bthirion) [inline doc/glossary.rst:115]:
```suggestion
        Mutli-Voxel Pattern Analysis. This is the way supervised learning methods are called in the field of brain imaging.
```

File: doc/glossary.rst [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+        `Searchlight analysis`_ consists of scanning the brain with a searchlight. 
+        That is, a ball of given radius is scanned across the brain volume and the 
+        prediction accuracy of a classifier trained on the corresponding voxels is measured.
+
+    SpaceNet
+        `SpaceNet`_ is a decoder implementing spatial penalties which improve brain 
+        decoding power as well as decoder maps.
+
+    SPM
+        Statistical Parametric Mapping.
```
Comment (bthirion) [inline doc/glossary.rst:151]: This is both a conceptual framework and a software. Please provide pointers.

File: doc/glossary.rst [hunk 15]
```diff
@@ -0,0 +1,257 @@
...
+                variable that splits the observations into two groups, such as 
+                patients versus controls.
+
+        In neuroimaging research, supervised learning is typically used to derive an 
+        underlying cognitive process (e.g., emotional versus non-emotional theory of 
+        mind), a behavioral variable (e.g., reaction time or IQ), or diagnosis status 
+        (e.g., schizophrenia versus healthy) from brain images.
+
+    SVM
+        TODO.
```
Comment (bthirion) [inline doc/glossary.rst:178]: link to kslearn + 1 short sentence

File: doc/glossary.rst [hunk 16]
```diff
@@ -0,0 +1,257 @@
...
+
+    TR
+        Repetition time.
```
Comment (bthirion) [inline doc/glossary.rst:181]: maybe reuse BIDS definition here

**Comment by NicolasGensollen** [comment]
@bthirion Shall I merge this?

File: doc/glossary.rst [hunk 17]
```diff
@@ -0,0 +1,305 @@
...
+        General Linear Model. This is the name of the models traditionally fit
+        to fMRI data, where one linear model is fit to each voxel time course.
+
+    HRF
+    hrf
+        Haemodynamic response function. This is a temporal filter that converts
+        neural signals to hemodynamic signals observable with :term:`fMRI`.
+
+    ICA
+        `Independant component analysis`_ is a computational method for separating
```
Comment (bthirion) [inline doc/glossary.rst:114]:
```suggestion
        `Independent component analysis`_ is a computational method for separating
```

File: doc/glossary.rst [hunk 18]
```diff
@@ -0,0 +1,305 @@
...
+        False positive rate correction. This refers to the methods employed to
+        correct false positive rates such as the Bonferroni correction which
+        divides the significance level by the number of comparisions made.
+
+    FREM
+        `FREM`_ means "Fast ensembling of REgularized Models". It uses an implicit
+        spatial regularization through fast clustering and aggregates a high 
+        number of estimators trained on various splits of the training set, thus 
+        returning a very robust decoder at a lower computational cost than other 
+        spatially regularized methods.
```
Comment (bthirion) [inline doc/glossary.rst:92]: I think that you should cite the paper: https://www.sciencedirect.com/science/article/abs/pii/S1053811917308182

File: doc/glossary.rst [hunk 19]
```diff
@@ -0,0 +1,305 @@
...
+
+    Neurovault
+        `Neurovault`_ is a public repository of unthresholded statistical maps, 
+        parcellations, and atlases of the human brain.
+
+    parcellation
+    parcellations
+    brain parcellation
+    brain parcellations
+        TODO.
```
Comment (bthirion) [inline doc/glossary.rst:140]: Can we take care of the remaining TODO's ? Do you need help for that ?

File: doc/glossary.rst [hunk 20]
```diff
@@ -0,0 +1,305 @@
...
+.. _`Family-wise error rate`:
+    https://en.wikipedia.org/wiki/Family-wise_error_rate
+
+.. _`FREM`:
+    https://nilearn.github.io/decoding/frem.html
+
+.. _`functional connectome`:
+    https://nilearn.github.io/connectivity/functional_connectomes.html
+
+.. _`Independant component analysis`:
```
Comment (bthirion) [inline doc/glossary.rst:271]:
```suggestion
.. _`Independent component analysis`:
```

**Review by bthirion — COMMENTED** [review]
There remain a couple of typos/missing definitions.

**Comment by emdupre** [comment]
A general question: Could we have multiple terms resolve to the same identifier, rather than mapping multiple terms to a single definition ? Looking [at the most recent artifacts](https://6883-1235740-gh.circle-artifacts.com/0/doc/_build/html/glossary.html), I still find it a bit confusing to have e.g. `BOLD`, `Bold`, and `bold` all listed for a single definition.

I think this mapping should be possible if we follow the second convention [shown here](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/glossary.html#link-a-term-to-its-a-glossary-entry). That is, by using:

```
:term:`BOLD<BOLD>`
:term:`Bold<BOLD>`
:term:`bold<BOLD>`
```

in text, all instances would all map to the single `BOLD` term in the glossary.

File: doc/glossary.rst [hunk 21]
```diff
@@ -0,0 +1,304 @@
...
+        `FREM`_ means "Fast ensembling of REgularized Models". It uses an implicit
+        spatial regularization through fast clustering and aggregates a high 
+        number of estimators trained on various splits of the training set, thus 
+        returning a very robust decoder at a lower computational cost than other 
+        spatially regularized methods. 
+
+    functional connectivity
+    functional connectome
+        A `functional connectome`_ is a set of connections representing brain 
+        interactions between regions.
```
Comment (emdupre) [inline doc/glossary.rst:89]:
```suggestion
    functional connectivity
        Functional connectivity is a measure of the similarity of the response patterns in two or more regions.
        A `functional connectome`_ is the set of connections representing brain interactions as measured by functional connectivity between regions.
```

File: doc/glossary.rst [hunk 22]
```diff
@@ -0,0 +1,304 @@
...
+        methods are called in the field of brain imaging.
+
+    Neurovault
+        `Neurovault`_ is a public repository of unthresholded statistical maps, 
+        parcellations, and atlases of the human brain.
+
+    parcellation
+    brain parcellation
+        Act of dividing the brain into smaller regions, i.e. parcels, in order to
+        gain a better understanding of complex human behaviors.
```
Comment (emdupre) [inline doc/glossary.rst:130]:
```suggestion
    parcellation
        Act of dividing the brain into smaller regions, i.e. parcels. Parcellations can be defined by many
        different criteria including anatomical or functional characteristics. Parcellations can either be composed of
        "hard" deterministic parcels with no overlap between individual regions or "soft" probabilistic parcels with
        a non-zero probability of overlap. 
```

**Review by emdupre — COMMENTED** [review]
Thanks, @NicolasGensollen ! The updated artifacts look much better IMHO.

Two suggestions to consolidate remaining "duplicate" terms and lightly update their definitions !

**Comment by NicolasGensollen** [comment]
Thanks a lot for the great suggestions @emdupre ! :+1:

**Review by emdupre — APPROVED** [review]
The [newest artifacts](https://6952-1235740-gh.circle-artifacts.com/0/dev/glossary.html) LGTM ! Excited to see this added :rocket: 

Thank you, @NicolasGensollen !

**Review by bthirion — APPROVED** [review]
LGTM, many thx !

**Comment by NicolasGensollen** [comment]
Thanks a lot for the reviews @emdupre @bthirion !
Merging! :tada:

**Pull request merged**